### PR TITLE
Create artifact to install IIS

### PR DESCRIPTION
**Pull request by workleodig**

### CODE REVIEW

File: Artifacts/windows-iis/artifact.ps1 [hunk 1]
```diff
@@ -0,0 +1,34 @@
+$artifactLogFolder = Join-path -path $env:ProgramData -childPath 'DTLArt_IIS'
+New-Item -ItemType Directory -Force -Path $artifactLogFolder
+
+$winVerString = (Get-WmiObject Win32_OperatingSystem).Caption
+Write-Output "Detected Windows Version: $winVerString"
+$scriptFolder = Split-Path $Script:MyInvocation.MyCommand.Path
+
+if ($winVerString -like 'Microsoft Windows Server 2008 R2*' -or $winVerString -like 'Microsoft Windows Server 2012*'){
+    Write-Output 'Installing IIS on Windows Server 2008 or 2012'
+	$localLog = Join-Path $artifactLogFolder 'IISArtifact.log'
```
Comment (leovms) [inline Artifacts/windows-iis/artifact.ps1]: Make sure the file is properly indented. Ideally, convert any Tab characters with Spaces.

File: Artifacts/windows-iis/artifact.ps1 [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    $pkgMgrExe = join-path $env:windir system32\pkgmgr.exe
+	$packages = 'IIS-WebServerRole;IIS-WebServer;IIS-IIS6ManagementCompatibility;IIS-Metabase;IIS-WMICompatibility;IIS-LegacyScripts;IIS-LegacySnapIn'
+    $pkgMgrArgs = "/l:$localLog /iu:$packages"
+    $pkgMgrProcess = Start-Process -FilePath $pkgMgrExe -ArgumentList $pkgMgrArgs -Wait -PassThru
+
+    if($pkgMgrProcess.ExitCode -eq 0){
```
Comment (leovms) [inline Artifacts/windows-iis/artifact.ps1]: Also check for reboot request scenarios (i.e. 3010).

File: Artifacts/windows-iis/artifact.ps1 [hunk 3]
```diff
@@ -0,0 +1,34 @@
+$artifactLogFolder = Join-path -path $env:ProgramData -childPath 'DTLArt_IIS'
+New-Item -ItemType Directory -Force -Path $artifactLogFolder
+
+$winVerString = (Get-WmiObject Win32_OperatingSystem).Caption
+Write-Output "Detected Windows Version: $winVerString"
+$scriptFolder = Split-Path $Script:MyInvocation.MyCommand.Path
+
+if ($winVerString -like 'Microsoft Windows Server 2008 R2*' -or $winVerString -like 'Microsoft Windows Server 2012*'){
+    Write-Output 'Installing IIS on Windows Server 2008 or 2012'
```
Comment (leovms) [inline Artifacts/windows-iis/artifact.ps1]: 2008 R2?

File: Artifacts/windows-iis/artifact.ps1 [hunk 4]
```diff
@@ -0,0 +1,34 @@
+$artifactLogFolder = Join-path -path $env:ProgramData -childPath 'DTLArt_IIS'
```
[the remark this answers is not in the document]
Comment (workleodig) [reply Artifacts/windows-iis/artifact.ps1]: ErrorActionPreference is set to Stop in artifact-main.ps1 so the artifact fails for any Write-Error calls

File: Artifacts/windows-iis/artifact.ps1 (hunk 3 above)
Comment (workleodig) [reply Artifacts/windows-iis/artifact.ps1]: Fixed

File: Artifacts/windows-iis/artifact.ps1 (hunk 1 above)
Comment (workleodig) [reply Artifacts/windows-iis/artifact.ps1]: done

File: Artifacts/windows-iis/artifact.ps1 (hunk 2 above)
Comment (workleodig) [reply Artifacts/windows-iis/artifact.ps1]: Added a message to the output if we encounter 3010

File: Artifacts/windows-iis/artifact.ps1 [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    $pkgMgrExe = join-path $env:windir system32\pkgmgr.exe
+    $packages = 'IIS-WebServerRole;IIS-WebServer;IIS-IIS6ManagementCompatibility;IIS-Metabase;IIS-WMICompatibility;IIS-LegacyScripts;IIS-LegacySnapIn'
+    $pkgMgrArgs = "/l:$localLog /iu:$packages"
+    $pkgMgrProcess = Start-Process -FilePath $pkgMgrExe -ArgumentList $pkgMgrArgs -Wait -PassThru
+
+    if($pkgMgrProcess.ExitCode -eq 0){
+        Write-Output 'Successfully installed IIS'
+    }
+    elseif($pkgMgrProcess.ExitCode -eq 3010){
+        Write-Output 'IIS install is finished. A reboot is required'
```
Comment (leovms) [inline Artifacts/windows-iis/artifact.ps1]: Minor, missing period after `A reboot is required`.

File: Artifacts/windows-iis/artifact.ps1 (hunk 4 above)
[the remark this answers is not in the document]
Comment (leovms) [reply Artifacts/windows-iis/artifact.ps1]: Yes, I'm just asking to make sure. So, to ensure, for example, I always use `throw`, but all good here.

File: Artifacts/windows-iis/artifact.ps1 (hunk 5 above)
Comment (workleodig) [reply Artifacts/windows-iis/artifact.ps1]: 😄

**Pull request merged**